avocado_vt/loader.py

```diff
@@ -79,6 +79,34 @@ def arch_listing(config, guest_name_parser=None):
     LOG.debug("")
 
 
+def get_vt_test_tags(vt_params):
+    """
+    Get the test categories based on a `tags` VT parameter.
+
+    :param vt_params: currently parsed VT parameters
+    :type vt_params: dict
+    :returns: test categories based on a `tags` vt param
+    :type: dict
+    """
+    result = {}
+    # XXX: Using whitespace as the delimiter is not a standard
+    #      implementation of Avocado's tags facility, this can
+    #      function properly until whitespace being considered
+    #      as a valid character for the tag identifier.
```

Comment:
Trying to understand your added comment here: what do you mean by "this can function properly until whitespace being considered as a valid character for the tag identifier"? This sounds a bit like a TODO and I don't think the plan is for Avocado to add support for them. Unless you are sure it is their limitation in which case opening a trackable issue might be better. There are some inherent differences between VT and pure Avocado so the standard usage of spaces for lists in VT shouldn't necessarily bother us too much and be changed to avocado-style or vice versa.

Reply:
In fact, the tags resolver [1] does support whitespace characters. For example, just by changing the delimiter to comma and assigning the `super fast` tag to the boot test, users can filter the test correctly.

```
$ avocado list --filter-by-tags='super fast'
VT io-github-autotest-qemu.boot
```

That's because the limitation is only applied to the parser in Avocado [2]. I cannot ensure Avocado would not revise the tags syntax forever, that's why I have the concern. And I said, Avocado should provide something like `tags.parse_tags` by the current implementation, if that's true then I have no chance to pick the delimiter.

Also, the truth is there is no built-in list type support in Cartesian config, "values are strings" [3]. One only can do serialization/deserialization to make the conversion between strings and lists, and the implementation of `str.split()` is too simple to use for deserialization.

[1] https://github.com/avocado-framework/avocado/blob/4a6641ab0b7c3f0bb1c568d2929d3955ceda3aba/avocado/core/tags.py#L19-L43
[2] https://github.com/avocado-framework/avocado/blob/4a6641ab0b7c3f0bb1c568d2929d3955ceda3aba/avocado/core/safeloader.py#L232
[3] https://avocado-vt.readthedocs.io/en/latest/CartesianConfig.html#keys-and-values

Reply:
> In fact, the tags resolver [1] does support whitespace characters. For example, just by changing the delimiter to comma and assigning the `super fast` tag to the boot test, users can filter the test correctly.
> 
> ```
> $ avocado list --filter-by-tags='super fast'
> VT io-github-autotest-qemu.boot
> ```
> 
> That's because the limitation is only applied to the parser in Avocado [2].

So perhaps it is better in the above comment to mention specifically "the limitation is only applied to the parser in Avocado" instead of talking generally about the "Avocado's tags facility". 

> And I said, Avocado should provide something like `tags.parse_tags` by the current implementation, if that's true then I have no chance to pick the delimiter.

It would still be great to track somewhere where you could express your comments for the long term, e.g. an Avocado issue. At least regarding this part.

> Also, the truth is there is no built-in list type support in Cartesian config, "values are strings" [3]. One only can do serialization/deserialization to make the conversion between strings and lists, and the implementation of str.split() is too simple to be used for deserialization.

I guess this depends on the interpretation. I consider the test parameters obtained via `Params` and in particular the `get_list` [function](https://github.com/avocado-framework/avocado-vt/blob/6142776eac1ea69f0b2461759fecf4899f91fb1f/virttest/utils_params.py#L139) to already achieve this. I have even called it with actual remote serialization using Pyro4, no manual parsing/deserialization of such a potential list was needed along the way.

Reply:
> So perhaps it is better in the above comment to mention specifically "the limitation is only applied to the parser in Avocado" instead of talking generally about the "Avocado's tags facility".

That is something unclarified by Avocado, I just wrote it based on my understanding, anyway I will update the comment per your suggestion if you think it is necessary.

> It would still be great to track somewhere where you could express your comments for the long term, e.g. an Avocado issue. At least regarding this part.

Yes, makes sense, let's track it by https://github.com/avocado-framework/avocado/issues/4452, thanks.